tests/testthat/test-pmcmc.R

```diff
@@ -291,7 +291,21 @@ test_that("can validate a vector of initial conditions", {
   expect_error(pmcmc_check_initial(c(x = 0.1, y = 0.2), dat$pars, 1),
                "Expected names of 'initial' to match parameters ('a', 'b')",
                fixed = TRUE)
-  expect_error(pmcmc_check_initial(c(-0.1, 0.2), dat$pars, 1),
+  a <- pmcmc_parameter("a", 1, integer = TRUE)
+  pars <- pmcmc_parameters$new(list(a), diag(1) * 0.1)
+  expect_error(pmcmc_check_initial(1.1, pars, 1),
+               "'initial' must be an integer but was not (chain 1)",
+               fixed = TRUE)
+  expect_error(pmcmc_check_initial(c(-1, 1), dat$pars, 1),
+               "'initial' is less than 'min' (0, 0) (chain 1)",
```

Comment:
Not totally convinced by these messages, any suggestions for more useful phrasing here?

Reply:
it's not clear here if the `0,0` refers to the expected min or the value given. Where things are guaranteed to be short we've thrown like:

```
value (X) is less than min (Y)
```

However, with these messages we could end up with really long strings if we're not careful - the sircovid model has ~60 parameters I think. It might be sufficient to report the name of the parameter that violates the constraint. It might also be worth detecting if _all_ chains fail the constraint, as that is likely in the default case and may simplify the reporting